mr/src/main/java/org/opensearch/hadoop/rest/commonshttp/auth/aws/AwsSigV4Signer.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.hadoop.rest.commonshttp.auth.aws;
+
+import java.security.InvalidKeyException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.temporal.ChronoField;
+import java.util.Collection;
+import java.util.Map;
+import java.util.TreeMap;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.codec.binary.Hex;
+
+import org.opensearch.hadoop.thirdparty.google.common.base.Charsets;
+import org.opensearch.hadoop.thirdparty.google.common.base.Joiner;
+import org.opensearch.hadoop.thirdparty.google.common.base.Optional;
+import org.opensearch.hadoop.thirdparty.google.common.base.Supplier;
+import org.opensearch.hadoop.thirdparty.google.common.base.Throwables;
+import org.opensearch.hadoop.thirdparty.google.common.collect.ImmutableList;
+import org.opensearch.hadoop.thirdparty.google.common.collect.ImmutableMap;
+import org.opensearch.hadoop.thirdparty.google.common.collect.Multimap;
+
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.utils.http.SdkHttpUtils;
+
+public class AwsSigV4Signer {
```

Comment:
It seems like `software.amazon.awssdk.auth.signer.Aws4Signer` already supports [1] that out of the box? 

```
Aws4SignerParams signerParams = Aws4SignerParams.builder()
                .awsCredentials(credentials.resolveCredentials())
                .signingName(this.signingServiceName)
                .signingRegion(signingRegion)
                .build();
        return Aws4Signer.create().sign(req.build(), signerParams);
```

[1] https://github.com/opensearch-project/opensearch-java/pull/324/files#diff-13b760e273e29a17fa88085ec653c7e9324e23dea21ff90c718adb6b9f25f3b0R385

Reply:
Thank you for this. I'll update my PR and try the shaded JAR route. It did not work previously but I'll give it another try.